glslc/src/file_includer.h

```diff
@@ -34,22 +36,36 @@ class FileIncluder : public shaderc::CompileOptions::IncluderInterface {
  public:
   FileIncluder(const shaderc_util::FileFinder* file_finder)
       : file_finder_(*file_finder) {}
-  shaderc_includer_response* GetInclude(const char* filename) override;
-  void ReleaseInclude(shaderc_includer_response* data) override;
+
+  // Resolves a requested source file of a given type from a requesting
+  // source into a shaderc_include_result whose contents will remain valid
+  // until it's released.
+  shaderc_include_result* GetInclude(const char* requested_source,
+                                     shaderc_include_type type,
+                                     const char* requesting_source,
+                                     size_t include_depth) override;
+  // Releases an include result.
+  void ReleaseInclude(shaderc_include_result* include_result) override;
+
+  // Returns a reference to the member storing the set of included files.
   const std::unordered_set<std::string>& file_path_trace() const {
-    return source_files_used_;
+    return included_files_;
   };
 
  private:
   // Used by GetInclude() to get the full filepath.
   const shaderc_util::FileFinder& file_finder_;
-  // Only one response needs to be kept alive due to the implementation of
-  // libshaderc's InternalFileIncluder::include_delegate, which make copies for
-  // the full path and content.
-  shaderc_includer_response response_;
-  std::vector<char> file_content_;
-  std::string file_full_path_;
-  std::unordered_set<std::string> source_files_used_;
+
+  // The full path and content of a source file.
+  struct FileInfo {
+    const std::string full_path;
+    std::vector<char> contents;
+  };
+
+  // The set of full paths of included files.
+  std::unordered_set<std::string> included_files_;
+  // Guards access to the included_files_.
+  std::mutex cache_mutex_;
```

Comment:
For consistency and safety shouldn't we mutex-guard the *read* of the member variable as well, via file_path_trace?
(And in that case since it's not just a trivial accessor, the read method name should be GetIncludedFiles since it does non-trivial processing.)